### PR TITLE
Loosen `azure-storage-blob` pin

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dhirschfeld** [comment]
@conda-forge-admin, please rerender

**Comment by dhirschfeld** [comment]
In the upstream repo it appears all pinning was removed in `0.7.3`:
* https://github.com/dask/adlfs/blame/v0.7.3/setup.py#L35

...and then a minimum pin of `12.5.0`  added in master:
* https://github.com/dask/adlfs/blame/master/setup.py#L35

So, following upstream, there's no reason this shouldn't work with the latest `12.8.0` version of `azure-storage-blob`

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ requirements:
     - azure-core >=1.5.0
     - azure-datalake-store >=0.0.46,<0.1
     - azure-identity
-    - azure-storage-blob >=12.5.0,<12.7.0
+    - azure-storage-blob >=12.5.0,<12.9.0
```
Comment (TomAugspurger) [inline recipe/meta.yaml:26]: In upstream there isn't an upper bound: https://github.com/dask/adlfs/blob/e2a3807b8748a48eec366730a6d6ed60e4192cf4/setup.py#L35. Just match that?

**Review by TomAugspurger — COMMENTED** [review]
Thanks for the update! Question on the bound for azure-storage-blob.

File: recipe/meta.yaml (hunk 1 above)
Comment (dhirschfeld) [reply recipe/meta.yaml:26]: I was being conservative as I figure it's easier to put in a PR to bump the pin than it is to patch the repodata to add a pin after the fact.

Happy to make the change to follow upstream though...

**Review by TomAugspurger — APPROVED** [review]
Thanks! I'll merge when CI passes.

**Comment by dhirschfeld** [comment]
Thanks Tom! :+1:

**Pull request merged**